### PR TITLE
Add the TextDirection to getRectsForRange

### LINKED ISSUE #1491: Considering moving examples to own repo?
Now that we have our own org, would it make sense to move examples out of engine and into their own org?

Or are things moving too fast right now...

### PR DESCRIPTION
**Pull request by abarth**
We need this in order to draw the cursor correctly.

Fixes #1491

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: sky/engine/core/rendering/RenderText.cpp [hunk 1]
```diff
@@ -220,21 +221,17 @@ void RenderText::absoluteRectsForRange(Vector<IntRect>& rects, unsigned start, u
...
             FloatRect r = box->calculateBoundaries();
-            if (useSelectionHeight) {
-                LayoutRect selectionRect = box->localSelectionRect(start, end);
-                r.setHeight(selectionRect.height().toFloat());
-                r.setY(selectionRect.y().toFloat());
-            }
-            rects.append(localToAbsoluteQuad(r, 0).enclosingBoundingBox());
+            boxes.emplace_back(localToAbsoluteQuad(r).enclosingBoundingBox(), box->direction());
         } else {
             // FIXME: This code is wrong. It's converting local to absolute twice. http://webkit.org/b/65722
```
Comment (Hixie) [inline sky/engine/core/rendering/RenderText.cpp]: this comment seems wrong? the webkit bug has general confusion too...

File: sky/engine/core/rendering/RenderText.cpp [hunk 2]
```diff
@@ -220,21 +221,17 @@ void RenderText::absoluteRectsForRange(Vector<IntRect>& rects, unsigned start, u
...
-            FloatRect rect = localQuadForTextBox(box, start, end, useSelectionHeight);
+            FloatRect rect = localQuadForTextBox(box, start, end, false);
```
Comment (Hixie) [inline sky/engine/core/rendering/RenderText.cpp]: say what the false means?

**Comment by Hixie** [comment]
lgtm

File: sky/engine/core/rendering/RenderText.cpp (hunk 1 above)
Comment (abarth) [inline sky/engine/core/rendering/RenderText.cpp]: Yeah, I think the comment is wrong.  I'll remove it.

**Pull request merged**